pageserver/src/tenant/par_fsync.rs

```diff
@@ -19,14 +21,8 @@ fn parallel_worker(paths: &[PathBuf], next_path_idx: &AtomicUsize) -> io::Result
     Ok(())
 }
 
-pub fn par_fsync(paths: &[PathBuf]) -> io::Result<()> {
```

Comment:
I think we can reuse the async path from spawn_blocking as long as we move fsync out of spawn_blocking of `create_delta_layer`, but I prefer to do this in future PRs.

Reply:
Not sure about how will that happen without Handle::block_on, but let's see it then :) Please remember the deadlock consideration. Resolving.

Reply:
> can we now remove rayon

not yet because IIRC we cannot use async functions inside `spawn_blocking`? once we move fsync out of spawn_blocking we can remove rayon.

> reusing the threads vs. always not reusing the threads

I would prefer reusing threads; once we switch to tokio we can automatically gain the benefit of running everything in tokio worker thread 🤪

But in this case, compaction / flush to L0 are background operations so the overhead does not matter that much. So I don't think we need to make the change right now.

Reply:
On mobile right now but doing a quick code search (probably in your pr branch) shows that we have no more usages for rayon.

Note the other part uses crossveam_utils thread scope, which is superceded by std::thread::scope.

Reply:
using std::thread::scope now